test/smoke/webhookoncall_test.go

```diff
@@ -0,0 +1,84 @@
+package smoke
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/target/goalert/expflag"
+	"github.com/target/goalert/test/smoke/harness"
+)
+
+type POSTDataOnCallUser struct {
+	ID   string
+	Name string
+	URL  string
+}
+
+type POSTDataOnCallNotification struct {
+	AppName      string
+	Type         string
+	Users        []POSTDataOnCallUser
+	ScheduleID   string
+	ScheduleName string
+	ScheduleURL  string
+}
+
+// TestWebhookOnCallNotification tests that the configured rule sends the intended notification to the webhook.
+func TestWebhookOnCallNotification(t *testing.T) {
+	t.Parallel()
+
+	ch := make(chan POSTDataOnCallNotification)
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var alert POSTDataOnCallNotification
+
+		data, err := io.ReadAll(r.Body)
+		require.Nil(t, err)
```

Comment:
Use `assert.*` in here -- `require.*` isn't allowed in background goroutines (like that of an HTTP request handler) since it uses `panic` internally.